bridges/TwitterBridge.php

```diff
@@ -251,104 +272,136 @@ public function collectData(){
 			}
 		}
 
+		// Filter out unwanted tweets
+		foreach ($data as $tweet) {
+			// Filter out retweets to remove possible duplicates of original tweet
+			switch($this->queriedContext) {
+			case 'By keyword or hashtag':
+				if (isset($tweet->retweeted_status) && substr($tweet->full_text, 0, 4) === 'RT @') {
+					continue 2;
+				}
+				break;
+			}
+			$tweets[] = $tweet;
+		}
+
 		$hidePictures = $this->getInput('nopic');
 
-		$promotedTweetIds = array_reduce($data->timeline->instructions[0]->addEntries->entries, function($carry, $entry) {
-			if (!isset($entry->content->item)) {
-				return $carry;
-			}
-			$tweet = $entry->content->item->content->tweet;
-			if (isset($tweet->promotedMetadata)) {
-				$carry[] = $tweet->id;
-			}
-			return $carry;
-		}, array());
+		// $promotedTweetIds = array_reduce($data->timeline->instructions[0]->addEntries->entries, function($carry, $entry) {
+		//    if (!isset($entry->content->item)) {
+		//        return $carry;
+		//    }
```

Comment:
You can remove this including commented lines. Promoted tweets does not seem to appear.